platform/solr/solr-factory-impl/src/main/java/org/codice/solr/factory/impl/SolrCloudClientFactory.java

```diff
@@ -63,40 +77,272 @@
+            .withDelay(100, TimeUnit.MILLISECONDS)
+            .withMaxDuration(3, TimeUnit.MINUTES)
+            .retryOn(RemoteSolrException.class);
+    CollectionAdminResponse response =
+        Failsafe.with(retryPolicy)
+            .get(
+                () ->
+                    CollectionAdminRequest.createAlias(alias, aliasedCollections).process(client));
+
+    if (response.getErrorMessages() != null && response.getErrorMessages().size() != 0) {
       LOGGER.warn(
-          "Cannot create SolrCloud client without Zookeeper host list system property [solr.cloud.zookeeper] being set.");
-      throw new IllegalStateException("system property 'solr.cloud.zookeeper' is not configured");
+          "Failed to update alias [{}] with collections: [{}], this will cause queries to be inconsistent. Error: {}",
+          alias,
+          aliasedCollections,
+          response.getErrorMessages());
+    }
+    if (!isAliasReady(client, alias)) {
+
+      LOGGER.debug(
+          "Alias [{}] was not updated with new list of collections: [{}]", alias, collections);
+    } else {
+      LOGGER.trace(
+          "Alias [{}] updated with new list of collections: [{}]", alias, aliasedCollections);
     }
-    LOGGER.debug(
-        "Solr({}): Creating a SolrCloud client using Zookeeper hosts [{}]", core, zookeeperHosts);
-    return new SolrClientAdapter(core, () -> createSolrCloudClient(zookeeperHosts, core));
   }
 
   @VisibleForTesting
-  SolrClient createSolrCloudClient(String zookeeperHosts, String collection) {
+  SolrClient createSolrCloudClient(
+      String zookeeperHosts,
+      String collection,
+      String alias,
+      SolrCollectionConfiguration configuration) {
     try (final Closer closer = new Closer()) {
       CloudSolrClient client = closer.with(newCloudSolrClient(zookeeperHosts));
       client.connect();
 
+      String configurationName = collection;
+      List<SolrConfigurationData> configurationData = null;
       try {
-        uploadCoreConfiguration(collection, client);
-      } catch (SolrFactoryException e) {
-        LOGGER.debug("Solr({}): Unable to upload configuration to SolrCloud", collection, e);
-        return null;
-      }
+        if (!isAliasCollection(collection, client)) {
 
-      try {
-        createCollection(collection, client);
+          int requestedShard = shardCount;
+          int requestedReplica = replicationFactor;
+          int requestedMaxShardsPerNode = maximumShardsPerNode;
+
+          if (configuration != null) {
+            configurationName = configuration.getConfigurationName();
+            configurationData = configuration.getSolrConfigurationData();
+            requestedShard = configuration.getDefaultNumShards().orElse(shardCount);
+            requestedReplica = configuration.getDefaultNumReplica().orElse(replicationFactor);
+            requestedMaxShardsPerNode =
+                configuration.getDefaultMaxShardsPerNode().orElse(maximumShardsPerNode);
```

Comment:
❗ Best I can tell, there's no difference between the state variables in this class and the values getting pulled from the config object. Both in this class and in the configuration's impl, the **same exact** values are getting pulled from the **same exact** system properties, but in two different places. 

❓ Why did we need this?